rspace/src/test/scala/coop/rchain/rspace/history/HistoryRepositorySpec.scala

```diff
@@ -221,6 +236,44 @@ trait InMemoryHistoryRepositoryTestBase extends InMemoryHistoryTestBase {
   def rootRepository =
     new RootRepository[Task](inmemRootsStore)
 
+  def inMemStore = new Store[Task] {
+    val data: TrieMap[ByteBuffer, ByteBuffer] = TrieMap.empty
+
+    override def close(): Task[Unit] = Task.delay(())
+
+    override def get(key: Blake2b256Hash): Task[Option[BitVector]] = Task.delay {
+      data.get(key.bytes.toByteBuffer).map(BitVector(_))
+    }
+
+    override def get(key: ByteBuffer): Task[Option[BitVector]] = Task.delay {
+      data.get(key).map(BitVector(_))
+    }
+
+    override def get[T](
+        keys: Seq[Blake2b256Hash],
+        fromBuffer: ByteBuffer => T
+    ): Task[Seq[Option[T]]] = Task {
+      keys.map(k => get(k.bytes.toByteBuffer).runSyncUnsafe().map(b => fromBuffer(b.toByteBuffer)))
+    }
+
+    override def put(data: Seq[(Blake2b256Hash, BitVector)]): Task[Unit] = Task {
+      data.map { case (k, v) => put(k, v).runSyncUnsafe() }
+      ()
+    }
+
+    override def put(key: Blake2b256Hash, value: BitVector): Task[Unit] = Task {
+      data.put(key.bytes.toByteBuffer, value.toByteBuffer)
+    }
+
+    override def put(key: ByteBuffer, value: ByteBuffer): Task[Unit] = Task { data.put(key, value) }
+
+    override def put[T](keys: Seq[(Blake2b256Hash, T)], toBuffer: T => ByteBuffer): Task[Unit] =
+      Task {
+        keys.map { case (k, v) => put(k.bytes.toByteBuffer, toBuffer(v)).runSyncUnsafe() }
+        ()
+      }
+  }
+
```

Comment:
We have in-memory implementation for Store via KeyValueStoreManager.
https://github.com/rchain/rchain/blob/4b44814895d424253366a9d7ed11dacbfa6c4205/shared/src/main/scala/coop/rchain/store/InMemoryStoreManager.scala#L9

And to get Store from KeyValueStore.
https://github.com/rchain/rchain/blob/b38985d9d147d841a4cd75fb81bdb759849b0147/rspace/src/main/scala/coop/rchain/rspace/history/StoreFactory.scala#L14

Reply:
together with  https://github.com/rchain/rchain/pull/3204#discussion_r532149581